### ISSUE TITLE
Ability to turn of LEDs on switch through home assistant

### ISSUE THREAD

**Issue by AcidWizard** [ckarrie/ha-netgear-plus#39]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ckarrie/ha-netgear-plus/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

I own a Netgear GS308EPP and I have the LEDs disabled at night as my SO is very sensitive to light when sleeping. Right now i have to do this manually through the switch GUI but it seems possible through this integration. I do have some programming experience but no experience with home assistant so I am not sure how to add this functionality to your integration.

### Describe the solution you'd like

Have a switch in Home Assistant that enables or disables the LED's so this can be automated.

### Describe alternatives you've considered

Making a switch/automation/script in homeassistant to manually call the port_led.cgi endpoint, but i have not gotten this to work.

### Additional context

I noticed in the network tab on my browser that the page port_led.cgi was called with a hash(I assume the hashed password) and an option named portled. This option is set to 2 for turning the LED's off and set to 0 to turn the LED's on.

**Comment by AcidWizard** [comment]
I think the switch_leds function should probably look like this:


    def switch_leds(self, state: str) -> bool:
        """Switch leds on or off."""
        if state not in SWITCH_STATES:
            return False
            for template in self.switch_model.LED_CONFIG_TEMPLATES:
                url = template["url"].format(ip=self.host)
                data = {
                    "hash": self._client_hash,
                    "portled": 0 if state == "on" else 2,
                }
                resp = self._request("post", url, data=data)
                if resp.status_code == requests.codes.ok:
                    return True
        return False
        
Where LED_CONFIG_TEMPLATES is a member of the GS308EPP and contains:
```
LED_CONFIG_TEMPLATES: ClassVar = [
        {"method": "get", "url": "http://{ip}/port_led.cgi"}
]


But I am not sure how to implement the switch in homeassistant itself

**Comment by foxey** [comment]
Nice! The PoE power switch has some side effects. Does the LED switching form have additional fields as well?

**Comment by AcidWizard** [comment]
No, its just an ON/OFF switch which you press apply on that disables the LEDs on the RJ45 ports, the power and POE max LED stay on. The request just has the hash and portled values.

**Comment by foxey** [comment]
The latest pre-release of py-netgear-plus (v0.4.2rc2) has support for front panel LED switching.
Use `sw.turn_on_leds()` or `sw.turn_off_leds()` to operate them. The `sw.get_switch_infos` call also has an additional attribute `led_status`.

Please try it out and let me know if it works. Next steps is to update the HA integration.

**Comment by foxey** [comment]
Released in [v0.7.4](https://github.com/ckarrie/ha-netgear-plus/releases/tag/v0.7.4).

**Comment by raspisteve** [comment]
I can't see the switch in Home Assistant using GS316EP.

**Comment by foxey** [comment]
It only works for the GS308EP(P) models. The GS316EP(P) models have a different implementation that I didn´t have found time for to implement it yet.

PS @raspisteve thereś one little caveat: HA doesn´t update the switch when it is changed from the web UI: the status is in the metadata page and that one is cached. As the current implementation already fetches a number of pages (sometime causing slow update warnings in HA), I did´t find it worth the additional page fetch every 10 seconds to get updates not initiated from HA.
